src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -182,6 +182,14 @@ out diagnostic
 				return true;
 			}
 
+			if( type.IsTupleType ) {
+				info = ImmutableTypeInfo.CreateWithAllImmutableTypeParameters(
+					kind: ImmutableTypeKind.Total,

```

Comment:
There could theoretically be some funkiness in this zone given `readonly (object, int) m_foo = (new object(), 1);` is theoretically okay...

Decide if we want to care/support that down the road?